### PR TITLE
Bidirectional text rendering (RTL)

### PR DESCRIPTION
**Pull request by elsassph**
New text texture renderers with opt-in bidirectional layout capability. This allows rendering right-to-left scripts (Arabic, Hebrew) and combination of left-to-right.

- `TextTextureAdvancedRenderer` now extends the `TextTextureRenderer`, so only the needed code can be overridden,
- Text rendering tokenizer can bow be overridden for instance to better support Chinese or Thai (not included by default),
- By default bidirectional text isn't supported; it has to be enabled.

The API was designed for tree-shaking: if you don't need bidi support, it won't be included.

New implementation passes existing tests (with minor API change) and accuracy is confirmed by new Playwright tests doing pixel comparison between real HTML rendering and the new renderer:

image

```typescript
// RTL support
import { TextTexture, TextTokenizer } from '@lightningjs/core';
import { getBidiTokenizer } from '@lightningjs/core/bidiTokenizer';
TextTokenizer.setCustomTokenizer(getBidiTokenizer());
TextTexture.forceAdvancedRenderer = true; // only advanced renderer supports bidi layout
```

```typescript
// Chinese/Thai support
import type { TextTokenizer } from '@3rdparty/lightningjs-core';
import { loadDefaultSimplifiedChineseParser } from 'budoux';

const getSimplifiedChineseTokenizer = (): TextTokenizer.ITextTokenizerFunction => {
  const parser = loadDefaultSimplifiedChineseParser();
  return (text) => [{ tokens: parser.parse(text) }];
};

TextTokenizer.setCustomTokenizer(getSimplifiedChineseTokenizer(), true); 
// This Chinese tokenizer is very efficient but doesn't correctly tokenize English, so the second 
// `true` parameter hints `TextTokenizer` to handle 100% English text with the default tokenizer.
```

### CODE REVIEW

**Comment by anand-patel-ap** [comment]
Hi Team,
When can we expect this PR to be reviewed and merged?

**Comment by elsassph** [comment]
@anand-patel-ap I found some edge cases so I'm working on hardening the implementation and adding unit tests. You may need to build it yourself to try.

**Comment by anand-patel-ap** [comment]
Sure, While building i got few mirnor issues. PFA: 
image

File: src/textures/TextTextureRendererUtils.ts [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        nowrap: true,
+      };
+    case "ellipsis":
+      return {
+        suffix: maxLinesSuffix,
+        nowrap: true,
+      };
+    default:
+      return {
+        suffix: textOverflow || maxLinesSuffix,
```
[the remark this answers is not in the document]
Comment (elsassph) [reply src/textures/TextTextureRendererUtils.ts:217]: That is wrong

**Comment by Drulokia** [comment]
Sofar the code looks good to me.
For the tests on test 9 it seems it behaves different on the text-indent for canvas when the width is large enough that it fits on one line

afbeelding

**Comment by elsassph** [comment]
@Drulokia OK the reason is that the texture is wider than 2048px which I have set as arbitrary max value. 
Lightning has code showing assumption that it is a max value - for instance in `Texture.mjs`
```
        /**
         * The (maximum) expected texture source width. Used for within bounds determination while texture is not yet loaded.
         * If not set, 2048 is used by ElementCore.update.
         * @type {number}
         */
        this.mw = 0;
```

**Comment by anand-patel-ap** [comment]
Hi, @elsassph 
I am trying to use this rtl feature branch in my code and after build this i am getting this error. Build is success but when running the code it is giving the error.
image

**Comment by anand-patel-ap** [comment]
Hi @elsassph 
Any idea what could cause this text flickring issue? It looks like when i am updating the text it is showing some shadow or hilightColor and then showing the text. 
https://github.com/user-attachments/assets/366132dd-859f-4b98-a53e-2ca5faa12ab4
This isse wasn't there in the earlier lightningjs/core. But i update my core with this PR since it is coming.
